src/Compilers/Core/Portable/Syntax/InternalSyntax/ChildSyntaxList.Enumerator.cs

```diff
@@ -86,7 +88,7 @@ public bool MoveNext()
 
             public GreenNode Current
```

Comment:
Why is this not annotated? Seems like it can trivially return null.

Reply:
This can only return `null` when `Current` is called after `MoveNext` returns `false`. That is using an object in an invalid state. For those cases which fit into C# patterns, like `IDisposable`, we've been annotating the valid use state.

---
In reply to: [364899689](https://github.com/dotnet/roslyn/pull/40767#discussion_r364899689) [](ancestors = 364899689)

Reply:
Alright, sounds good.

---
In reply to: [364901015](https://github.com/dotnet/roslyn/pull/40767#discussion_r364901015) [](ancestors = 364901015,364899689)